react-validated-input-component/src/validated-input.jsx

```diff
@@ -16,16 +15,24 @@ class ValidatedInput extends React.Component {
 
   handleChange(event) {
     const value = event.target.value;
-    const name = event.target.name;
-    this.setState({ [name]: value });
+    const regex = /^(?=.*\d)(?=.*[A-Z])(?=.*[a-zA-Z]).+$/gm;
```

Comment:
This is really close it checks for lowercase, uppercase, and numbers. The only thing it doesn't check for is the special characters. Great job with it! Here is a good site for learning more about regex and testing regex:

https://regex101.com/